docs/blog/2020-01-30-why-gatsby-is-better-with-javascript/index.md

```diff
@@ -0,0 +1,60 @@
+tags:
+  - Gatsby
+  - JavaScript
+  - Performance
+  - Accessibility
+---
+
+Gatsby sites can work without JavaScript. Because Gatsby renders all the pages as plain HTML files, users can use the majority of a website's functionality without it. JavaScript is required only for additional interactive elements of an app.
+
+We have often been asked to add an option to not build or include JavaScript bundles in a site. However we believe that this will make Gatsby performance worse, not better, and this blog post aims to explain why we think this is the case. We will also cover how some future React features, like partial hydration, will improve Gatsby performance.
+
+## What is JavaScript used for in Gatsby sites?
+
+In a nutshell, a Gatsby site is the result of React [Server-Side Rendering](https://www.gatsbyjs.org/docs/glossary#server-side) each page to an HTML file. We prepare JavaScript bundles that include React and the components used on your pages, so after the user loads the HTML page, it loads the minimal bundle so React can [hydrate the HTML](https://www.gatsbyjs.org/docs/react-hydration) and become a Single-Page App. That means that adding interactivity to a Gatsby site is very easy and requires no additional steps - React just works as usual.
+
+In order to add interactivity to a page Gatsby has to ship JavaScript. Doing this in a traditional server-oriented framework or CMS like WordPress is more awkward - it becomes a separate part of application that isn’t connected to main rendering code. To make it worse, this often requires shipping libraries like jQuery (or it’s plugins). That makes even the initial bundle bigger than what Gatsby ships by default.
+
+But what about sites or pages where there is no client-side interactivity? For these, by including JavaScript, Gatsby still offers actual performance benefits. A Gatsby site is structured around the concept of “pages”. A page is an entry point in a Gatsby application, one can think of it as a single URL on the website. It's quite rare for a site to only have one page. More often, users visit multiple pages on a website. On a regular website a page transition would mean that the browser has to download a new html page and then all of the corresponding assets. A Gatsby site, however, will instead download a json file (called “page-data”) for the new page instead. Inside that file, Gatsby stores all the data that is needed to render the page. When JavaScript is enabled, after Gatsby downloads that file (and the component bundle needed to render this page, if it’s a new route), it will do a client-side transition to that page. Gatsby can preload those files based on the links that are visible in the viewport. In many cases that would mean that transition would feel snappier for the user, as there wouldn’t be a need of full page reload and preloaded files would be already available.
```

Comment:
```suggestion
But what about sites or pages where there is no client-side interactivity? Even for those pages, Gatsby offers performance benefits by including JavaScript. A Gatsby site is structured around the concept of “pages”. A page is an entry point in a Gatsby application, one can think of it as a single URL on the website. It's quite rare for a site to only have one page. More often, users visit multiple pages on a website. On a regular website a page transition would mean that the browser has to download a new HTML page and then all of the corresponding assets. A Gatsby site, however, will download a JSON file (called “page-data”) for the new page instead. Inside that file, Gatsby stores all the data that is needed to render the page. When JavaScript is enabled, after Gatsby downloads that file (and the component bundle needed to render this page, if it’s a new route), it will do a client-side transition to that page. Gatsby can preload those files based on the links that are visible in the viewport. In many cases that means the transition will feel snappier for the user, as there is no need to do a full page reload and preloaded files would already be available.
```